src/user/BFB_surface_forcing.F90

```diff
@@ -0,0 +1,326 @@
+  ! call MOM_error(FATAL, "User_buoyancy_surface_forcing: " // &
+  !   "User forcing routine called without modification." )
+
+  ! Allocate and zero out the forcing arrays, as necessary.  This portion is
+  ! usually not changed.
+  if (CS%use_temperature) then
+    call alloc_if_needed(fluxes%evap, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%lprec, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%fprec, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%lrunoff, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%frunoff, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%vprec, isd, ied, jsd, jed)
+
+    call alloc_if_needed(fluxes%sw, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%lw, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%latent, isd, ied, jsd, jed)
+    call alloc_if_needed(fluxes%sens, isd, ied, jsd, jed)
+  else ! This is the buoyancy only mode.
+    call alloc_if_needed(fluxes%buoy, isd, ied, jsd, jed)
+  endif
+
+
+  ! MODIFY THE CODE IN THE FOLLOWING LOOPS TO SET THE BUOYANCY FORCING TERMS.
+
+  if ( CS%use_temperature ) then
+    ! Set whichever fluxes are to be used here.  Any fluxes that
+    ! are always zero do not need to be changed here.
+    do j=js,je ; do i=is,ie
+      ! Fluxes of fresh water through the surface are in units of kg m-2 s-1
+      ! and are positive downward - i.e. evaporation should be negative.
+      fluxes%evap(i,j) = -0.0 * G%mask2dT(i,j)
+      fluxes%lprec(i,j) = 0.0 * G%mask2dT(i,j)
+
+      ! vprec will be set later, if it is needed for salinity restoring.
+      fluxes%vprec(i,j) = 0.0
+
+      !   Heat fluxes are in units of W m-2 and are positive into the ocean.
+      fluxes%lw(i,j) = 0.0 * G%mask2dT(i,j)
+      fluxes%latent(i,j) = 0.0 * G%mask2dT(i,j)
+      fluxes%sens(i,j) = 0.0 * G%mask2dT(i,j)
+      fluxes%sw(i,j) = 0.0 * G%mask2dT(i,j)
+    enddo ; enddo
+  else ! This is the buoyancy only mode.
+    do j=js,je ; do i=is,ie
+      !   fluxes%buoy is the buoyancy flux into the ocean in m2 s-3.  A positive
+      ! buoyancy flux is of the same sign as heating the ocean.
+      fluxes%buoy(i,j) = 0.0 * G%mask2dT(i,j)
+    enddo ; enddo
+  endif
+
+  if (CS%restorebuoy) then
+    if (CS%use_temperature) then
+      call alloc_if_needed(fluxes%heat_restore, isd, ied, jsd, jed)
+      !   When modifying the code, comment out this error message.  It is here
+      ! so that the original (unmodified) version is not accidentally used.
+      call MOM_error(FATAL, "User_buoyancy_surface_forcing: " // &
+        "Temperature and salinity restoring used without modification." )
+
+      rhoXcp = CS%Rho0 * fluxes%C_p
+      do j=js,je ; do i=is,ie
+        !   Set Temp_restore and Salin_restore to the temperature (in C) and
+        ! salinity (in PSU) that are being restored toward.
+        Temp_restore = 0.0
+        Salin_restore = 0.0
+
+        fluxes%heat_restore(i,j) = (G%mask2dT(i,j) * (rhoXcp * CS%Flux_const)) * &
```

Comment:
Heads up: I had one issue at this line because @MJHarrison-GFDL renamed fluxes%heat_restore to fluxes%heat_added in #336 but the units did not change so I've simply patched this file and finished the merge.